test/common/testlib.py

```diff
@@ -1147,31 +1147,33 @@ def filterOutput(self, test, failed, output):
         tries = getattr(test, "retryCount", 0)
         tries += 1
         setattr(test, "retryCount", tries)
+        # "output" is bytes, grab corresponding stream
+        out = (sys.version_info[0] >= 3) and sys.stdout.buffer or sys.stdout
```

Comment:
okay.  one mild nag, which is 100% mater of taste:

i'd write this as: `hasattr(sys.stdout, 'buffer')` with a comment about py3

Reply:
I factored out the `sys.version` check (which already existed in another place) into a new `_PY3` constant. #9571 will do that anyway, so I'll rebase that on top of this PR.